cache/in-memory/src/lib.rs

```diff
@@ -112,35 +169,28 @@ struct InMemoryCacheRef {
 /// This is an implementation of a cache designed to be used by only the
 /// current process.
 ///
-/// # Public Immutability
+/// # Design and Performance
 ///
 /// The defining characteristic of this cache is that returned types (such as a
-/// guild or user) do not use locking for access. Although the internals of the
-/// cache use asynchronous locking for mutability, the returned types themselves
-/// are immutable. If a user is retrieved from the cache, an `Arc<User>` is
-/// returned. If a reference to that user is held but the cache updates the
-/// user, the reference held by you will be outdated, but still exist.
+/// guild or user) do not use locking for access. The internals of the cache use
+/// a lock-free concurrent map for mutability and the returned types themselves
```

Comment:
Lock-free is not really true as of dashmap 3, it is first dashmap 4 that will be lock-free. So maybe just concurrent?